.pre-commit-config.yaml

```diff
@@ -0,0 +1,30 @@
+# See https://pre-commit.com for more information
+# See https://pre-commit.com/hooks.html for more hooks
+# See https://github.com/pre-commit/pre-commit-hooks/blob/master/.pre-commit-config.yaml for an example with more hooks
+
+repos:
+- repo: https://github.com/pre-commit/pre-commit-hooks
+  rev: v4.2.0
+  hooks:
+  - id: check-ast                  # Simply check whether the files parse as valid python
+  - id: check-yaml                 # Attempts to load all yaml files to verify syntax
+
+  - id: check-added-large-files    # Prevent giant files from being committed
+  - id: check-docstring-first      # Checks a common error of defining a docstring after code
+  - id: check-merge-conflict       # Check for files that contain merge conflict strings
+  - id: check-vcs-permalinks       # Ensures that links to vcs websites are permalinks
+  - id: detect-private-key         # Detects the presence of private keys
+  - id: end-of-file-fixer          # Ensures that a file is either empty, or ends with one newline
+  - id: trailing-whitespace        # This hook trims trailing whitespace
+  - id: mixed-line-ending          # Replaces or checks mixed line ending
+  - id: check-shebang-scripts-are-executable  # Checks that scripts with shebangs are executable
+
+- repo: https://github.com/PyCQA/flake8
+  rev: 4.0.1
+  hooks:
+  - id: flake8
+
+- repo: https://github.com/pre-commit/mirrors-autopep8
+  rev: v1.5.7
+  hooks:
+  - id: autopep8
```

Comment:
why not adding also a hook for mypy?

Reply:
We can. @ant0nsc feared that it might make things slow.

Reply:
As below, I suggest leaving it out for now and adding it later on if we want to.